packages/babel-helper-wrap-function/src/index.js

```diff
@@ -3,23 +3,21 @@ import nameFunction from "@babel/helper-function-name";
 import template from "@babel/template";
 import * as t from "@babel/types";
 
-const buildWrapper = template(`
-  (() => {
+const buildExpressionWrapper = template.expression(`
+  (function () {
     var REF = FUNCTION;
     return function NAME(PARAMS) {
       return REF.apply(this, arguments);
     };
-  })
+  })()
 `);
 
-const namedBuildWrapper = template(`
-  (() => {
-    var REF = FUNCTION;
-    function NAME(PARAMS) {
-      return REF.apply(this, arguments);
-    }
-    return NAME;
-  })
+const buildDeclarationWrapper = template(`
+  function NAME(PARAMS) { return REF.apply(this, arguments); }
+  function REF() {
+    REF = FUNCTION;
```

Comment:
I suppose it's ok for the `FUNCTION` to be unnamed. Hopefully nobody uses `arguments.callee`...

I wonder, how would this behave if the wrapped function itself writes to its own identifier? Would it just replace the trampoline function? In any case, looks safe to me.

Reply:
> I wonder, how would this behave if the wrapped function itself writes to its own identifier? Would it just replace the trampoline function? In any case, looks safe to me.

It works, it replaces the `function NAME(PARAMS) { return REF.apply(this, arguments); }` function :v: 

https://babeljs.io/repl/build/6145/#?babili=false&browsers=&build=&builtIns=false&code_lz=IYZwngdgxgBAZgV2gFwJYHsLwgCgJQwDeAUDNjALwyiSyIoZZwQBM-RMATgKbIKdYWAbhgBfUl178sARiHFxxAG7BOkkAgA2yEJRgBtALrziNaPCRQ0mLknYkyKtZxl7gAd2Cpk2fPMeqXCxunt6-eCZkwJrcnMg4-lwyADRBMIYwANQwAESUFFSJKTDBGQD8OREKxJx2EUA&debug=false&circleciRepo=&evaluate=true&lineWrap=true&presets=es2017&prettier=false&targets=Node-8.9&version=7.0.0-beta.34%2Bpr.6984

Reply:
The lack of var preceding "REF = FUNCTION;" seems to cause #7134

Reply:
There shouldn't be var, because it needs to replace the REF function.

Reply:
Ok. That was my bad then. I believe I have found the actual problem here https://github.com/facebook/regenerator/issues/225